google-cloud-storage/test/google/cloud/storage/retry/utils.rb

```diff
@@ -0,0 +1,464 @@
+  CONF_TEST_FILE_CONTENT = "my-test-file"
+  CONF_TEST_FILE_PATH = "my-test-file.txt"
+
+  ########################################################################################################################################
+  ### Method Invocation Mapping ##########################################################################################################
+  ########################################################################################################################################
+
+  # Method invocation mapping is a map whose keys are a string describing a standard
+  # API call (e.g. storage.objects.get) and values are a list of functions which
+  # wrap library methods that implement these calls. There may be multiple values
+  # because multiple library methods may use the same call (e.g. get could be a
+  # read or just a metadata get).
+
+  def self.get
+  {
+    "storage.bucket_acl.list" => [:list_bucket_acls],
+    "storage.buckets.delete" => [:delete_bucket],
+    "storage.buckets.get" => [
+      :get_bucket,
+      :bucket_reload
+    ],
+    "storage.buckets.getIamPolicy" => [:get_bucket_policy],
+    "storage.buckets.insert" => [:insert_bucket],
+    "storage.buckets.list" => [:list_buckets],
+    "storage.buckets.lockRetentionPolicy" => [:bucket_lock_retention_policy],
+    "storage.buckets.testIamPermissions" => [:test_bucket_permissions, :test_permissions],
+    "storage.default_object_acl.list" => [:list_default_acls],
+    "storage.hmacKey.create" => [:create_hmac_key],
+    "storage.hmacKey.delete" => [:delete_hmac_key],
+    "storage.hmacKey.get" => [
+      :get_hmac_key,
+      :hmac_key_reload
+    ],
+    "storage.hmacKey.list" => [:list_hmac_keys],
+    "storage.notifications.delete" => [:delete_notification],
+    "storage.notifications.get" => [
+      :get_notification,
+      :bucket_get_notification
+    ],
+    "storage.notifications.list" => [
+      :list_notifications,
+      :bucket_notifications
+    ],
+    "storage.object_acl.list" => [:list_file_acls],
+    "storage.object_acl.delete" => [:delete_file_acl],
+    "storage.object_acl.insert" => [:insert_file_acl],
+    "storage.objects.get" => [
+      :get_object,
+      :get_file,
+      :download_file
+    ],
+    "storage.objects.insert" => [
+      :insert_object
+    ],
+    "storage.objects.compose" => [:compose_file],
+    "storage.objects.delete" => [:delete_file],
+    "storage.objects.list" => [:list_files],
+    "storage.objects.patch" => [:patch_file],
+    "storage.objects.rewrite" => [:rewrite_file],
+    "storage.serviceaccount.get" => [:project_service_account],
+    "storage.buckets.patch" => [
+      :patch_bucket,
+      :bucket_acl_public,
+      :bucket_acl_private,
+      :default_acl_owner_full,
+    ],
+    "storage.buckets.setIamPolicy" => [:set_bucket_policy],
+    "storage.hmacKey.update" => [:update_hmac_key],
+    "storage.resumable.upload" => [
+      :blob_upload_from_string,
```

Comment:
Also, I notice that I think these still need to be implemented, correct? Fine to do it in a separate PR.

Reply:
I've implemented them as part of `storage.objects.insert` . If we need to implement more varieties of it, can be done in separate PR.